app/services/find_programme_activities.rb

```diff
@@ -17,11 +18,12 @@ def call(eager_load_parent: true)
       .includes(eager_load_associations)
       .order("created_at ASC")
 
-    programmes = if organisation.service_owner
-      programmes.all
-    else
-      programmes.where(extending_organisation_id: organisation.id)
-    end
-    programmes
+    return programmes if organisation.service_owner
+
+    programmes.where(extending_organisation_id: organisation.id).where(fund_query_condition(fund_id))
```

Comment:
If you're worried about this being too magical, another option would be to make the whole `.where` call conditional on having a `fund_id`, i.e.:

```rb
programmes = programmes.where(extending_organisation_id: organisation.id)
programmes = programmes.where(parent_id: fund_id) if fund_id.present?

programmes
```

Or, dynamically build the conditions, as in:

```rb
conditions = { extending_organisation_id: organisation.id }
conditions[:parent_id] = fund_id if fund_id.present?

programmes.where(conditions)
```
```

Reply:
For the first one, does Rails run just one query?

Reply:
None of these snippets execute a query -- `where` does not trigger a database request on its own. The query is only executed when you attempted to read the result, e.g. by calling `each` or `first`. So you can build a query by incrementally adding `where` calls to a chain, and it effectively merges their conditions into a single query.

Reply:
@jcoglan thanks - that is a super fundamentail bit of knowledge to learn!

Reply:
@jcoglan I could not get the first way to work at all, not sure what wasn't right, I've settled on the second way, I feel like it is the most approachable.